apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/DynamicLargeObjectApiLiveTest.java

```diff
@@ -0,0 +1,175 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jclouds.io.Payloads.newByteSourcePayload;
+import static org.testng.Assert.assertNotNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jclouds.io.Payloads;
+import org.jclouds.openstack.swift.v1.domain.ObjectList;
+import org.jclouds.openstack.swift.v1.domain.Segment;
+import org.jclouds.openstack.swift.v1.domain.SwiftObject;
+import org.jclouds.openstack.swift.v1.internal.BaseSwiftApiLiveTest;
+import org.jclouds.openstack.swift.v1.options.ListContainerOptions;
+import org.jclouds.utils.TestUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.io.ByteSource;
+
+@Test(groups = "live", testName = "DynamicLargeObjectApiLiveTest", singleThreaded = true)
+public class DynamicLargeObjectApiLiveTest extends BaseSwiftApiLiveTest {
+
+   private String defaultName = getClass().getSimpleName();
+   private String defaultContainerName = getClass().getSimpleName() + "Container";
+   private ByteSource megOf1s;
+   private ByteSource megOf2s;
+   private String objectName = "myObject";
+
+   @Test
+   public void testReplaceManifest() throws Exception {
+      for (String regionId : regions) {
+         assertReplaceManifest(regionId, defaultContainerName, defaultName);
+         uploadLargeFile(regionId);
+      }
+   }
+
+   @SuppressWarnings("deprecation")
+   @Test
+   public void uploadLargeFile(String regionId) throws IOException, InterruptedException {
+      List<Segment> segmentList = new ArrayList<Segment>();
+      int partNumber = 1;
+      int total_size = 0;
+      // configure the blobstore to use multipart uploading of the file
+      for (int i = partNumber; i < 3; partNumber++) {
+         String objName = String.format("%s/%s/%s", objectName, "dlo", partNumber);
+         String data = String.format("%s%s", "data", partNumber);
+         String etag = getApi().getDynamicLargeObjectApi(regionId, defaultContainerName).uploadPart(
+               defaultContainerName, objName, Payloads.newPayload(data), ImmutableMap.of("myfoo", "Bar"),
+               ImmutableMap.of("myfoo", "Bar"));
+         Segment s = new Segment(objName, etag, data.length());
+         assertNotNull(etag);
+         segmentList.add(s);
+         total_size += data.length();
+      }
+      String etagOfEtags = getApi().getDynamicLargeObjectApi(regionId, defaultContainerName).replaceManifest(
+            objectName, segmentList, ImmutableMap.of("MyFoo", "Bar"), ImmutableMap.of("MyFoo", "Bar"));
+
+      SwiftObject bigObject = getApi().getObjectApi(regionId, defaultContainerName).get(objectName);
+      assertThat(bigObject.getETag().equals(etagOfEtags));
```

Comment:
You misused `assertThat` -- all of these checks are no-ops.  You need to write `assertThat(...).isEqualTo` or `.isTrue`.